crates/context/src/lib.rs

```diff
@@ -307,18 +307,13 @@ impl ContextManager {
         blob_id: calimero_primitives::blobs::BlobId,
         source: calimero_primitives::application::ApplicationSource,
         version: Option<semver::Version>,
-        contract_app_id: Option<String>,
+        metadata: Option<String>,
```

Comment:
how about we take a `Vec<u8>` here instead of assuming a `String` (which is required to be utf8-encoded)?

also, an empty array can indicate absence of metadata.

`Vec::new` does not allocate, so there's no associated heap cost

Reply:
Updated